python/ql/src/semmle/python/objects/ObjectAPI.qll

```diff
@@ -617,6 +628,15 @@ class PythonFunctionValue extends FunctionValue {
         result = this.(PythonFunctionObjectInternal).getScope().getQualifiedName()
     }
 
+    override string descriptiveString() {
+        if this.getScope().isMethod()
+        then
+            exists(Class cls | this.getScope().getScope() = cls |
```

Comment:
I wonder if something can have `isMethod` be true, but not have no `Class` for its grandparent scope. That would be bad, since then this would have no `descriptiveString` result (and just silently fail, which we want to avoid). :thinking:

Reply:
Pretty sure it can't. `FunctionValue::getScope()` results in a `Function`, and `Function has the following two definitions:

```
override Scope getScope() { result = this.getEnclosingScope() }

    /** Whether this function is declared in a class */
    predicate isMethod() { exists(Class cls | this.getEnclosingScope() = cls) }
```

Reply:
Ah, good. :+1: